src/Kokkos_ArithTraits.hpp

```diff
@@ -427,6 +427,21 @@ class ArithTraits {
   //! The one value of T; the multiplicative identity.
   static KOKKOS_FORCEINLINE_FUNCTION T one ();
 
+  /// \brief True if this type T is capable of representing the
+  /// positive infinity as a distinct special value, as with
+  /// std::numeric_limits<T>::has_infinity.
+  static constexpr bool hasInfinity = false;
+
+  /// \brief Returns the special value "positive infinity", as
+  /// represented by the floating-point type T. Only meaningful if
+  /// KokkosArithTraits<T>::hasInfinity == true. Provides same
+  /// functionality as std::numeric_limits<T>::infinity().
+  ///
+  /// \note Would have liked to mark it as constexpr but then would
```

Comment:
Does Kokkos pass along `KOKKOS_CXX_STANDARD` or `KOKKOS_ENABLE_CXX14` to code as macros?  If so, you could select on that macro to make this `constexpr` (but note CUDA issues).

Reply:
What CUDA issues are your referring to?

Reply:
And I just grepped for `KOKKOS_CXX_STANDARD` and `KOKKOS_ENABLE_CXX` and it looks like `KOKKOS_CXX_STANDARD` is not passed along but `KOKKOS_ENABLE_CXX14` is.  I am not sure the complexity added is worth it at the moment unless there is a real need for it in which case we probably should consider make other member constants and member functions of `ArithTraits` constexp as well.

Reply:
This shows as outdated.  Not sure if you have seen it @mhoemmen.